org/mozilla/jss/provider/javax/crypto/JSSTokenKeyManager.java

```diff
@@ -0,0 +1,131 @@
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ * All rights reserved.
+ * END COPYRIGHT BLOCK */
+
+package org.mozilla.jss.provider.javax.crypto;
+
+import java.net.Socket;
+import java.security.KeyStore;
+import java.security.Principal;
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import javax.net.ssl.X509KeyManager;
+
+import org.mozilla.jss.CryptoManager;
+import org.mozilla.jss.NotInitializedException;
+import org.mozilla.jss.crypto.ObjectNotFoundException;
+import org.mozilla.jss.pkcs11.PK11Cert;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JSSTokenKeyManager implements JSSKeyManager {
+
+    final static Logger logger = LoggerFactory.getLogger(JSSCryptoKeyManager.class);
+
+    private KeyStore jks;
+    private CryptoManager cm;
+
+    public JSSTokenKeyManager(KeyStore jssKeyStore) {
+        jks = jssKeyStore;
+
+        try {
+            cm = CryptoManager.getInstance();
+        } catch (NotInitializedException nie) {
+            String msg = "CryptoManager reported as not initialized but have ";
+            msg += "a working KeyStore instance! " + nie.getMessage();
+            throw new RuntimeException(msg, nie);
+        }
+    }
+
+    @Override
+    public String chooseClientAlias(String[] keyTypes, Principal[] issuers, Socket socket) {
+        logger.debug("JSSKeyManager: chooseClientAlias() - not implemented");
+        return null;
+    }
+
+    @Override
+    public String chooseServerAlias(String keyType, Principal[] issuers, Socket socket) {
+        logger.debug("JSSKeyManager: chooseServerAlias() - not implemented");
+        return null; 
+    }
+
+    public org.mozilla.jss.crypto.X509Certificate getCertificate(String alias) {
+        try {
+            CryptoManager cm = CryptoManager.getInstance();
+            return (org.mozilla.jss.crypto.X509Certificate) jks.getCertificate(alias);
+        } catch (Exception e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public X509Certificate[] getCertificateChain(String alias) {
+
+        logger.debug("JSSKeyManager: getCertificateChain(" + alias + ")");
+
+        try {
+            org.mozilla.jss.crypto.X509Certificate cert = getCertificate(alias);
+
+            org.mozilla.jss.crypto.X509Certificate[] chain = cm.buildCertificateChain(cert);
+            logger.debug("JSSKeyManager: cert chain:");
+
+            Collection<org.mozilla.jss.pkcs11.PK11Cert> list = new ArrayList<>();
+            for (org.mozilla.jss.crypto.X509Certificate c : chain) {
+                if (!(c instanceof PK11Cert)) {
+                    logger.error("JSSKeyManager: Can't cast to PK11Cert: " + c);
+                }
```

Comment:
This is from the existing code, but it's probably not necessary since all classes implementing the `org.mozilla.jss.crypto.X509Certificate` are subclasses of `PK11Cert`.